crates/ark/src/modules/positron/system.R

```diff
@@ -21,3 +21,11 @@ has_x11 <- function() {
     out <- as.list(vapply(cats, Sys.getlocale, "string", USE.NAMES = TRUE))
     c(LANG = Sys.getenv("LANG"), out)
 }
+
+#' Reports a vector of environment variables for the R process.
+#' @returns Named character vector of env vars that start with the given regex.
+#' @export
+.ps.rpc.get_env_vars <- function(pattern) {
+    matches <- grep(pattern, names(Sys.getenv()), value = TRUE)
+    as.list(vapply(matches, Sys.getenv, "string"))
+}
```

Comment:
I wonder if we will want some default behaviour that just returns all the env vars. Maybe something like:

```suggestion
.ps.rpc.get_env_vars <- function(pattern = NULL) {
  ev <- as.list(Sys.getenv())
  m <- if (is.null(pattern)) TRUE else grep(pattern, names(ev))
  ev[m]
}
```

Reply:
I tested this out and it works well. 👍